frontend/src/components/common/form/ApplicationSelect.vue

```diff
@@ -0,0 +1,67 @@
+<script setup lang="ts">
+import { ref, type PropType, onMounted } from 'vue';
+import Dropdown, { type DropdownChangeEvent } from 'primevue/dropdown';
+import type { FamApplication } from 'fam-app-acsctl-api';
+
+const props = defineProps({
+    label: { type: String, required: false, default: 'Select application:' },
+    model: {
+        type: (Object as PropType<FamApplication>) || null,
+        required: true,
+    },
+    options: {
+        type: Array<FamApplication>,
+        default: [],
+        required: true,
+    },
+    placeholder: {
+        type: String,
+        required: false,
+        default: 'Choose an application',
+    },
+});
+
+const model = ref<FamApplication>();
+
+const emit = defineEmits(['onApplicationSelected']);
+
+onMounted(() => {
+    model.value = props.model;
+});
+
+const onApplicationSelected = async (e: DropdownChangeEvent) => {
+    emit('onApplicationSelected', e);
```

Comment:
Hi Nick, somehow I feel we're missing something, even though this component seems work fine. However, when we select application, we make the emit call, but your local state variable "model" is not changed. We only set the local model when on mounted to let it equals props.model. But `onApplicationSelected`, the emit method will update props.model, but there is no where to update the local model?

Reply:
Your suggestion does work, I'll fix it. Thanks.